app/assets/javascripts/miq_flash.js

```diff
@@ -27,6 +27,10 @@ function add_flash(msg, level, options) {
       throw("add_flash: unknown level: " + level);
   }
 
+  if (options.long_alert) {
```

Comment:
you should not depend on options here,
we can deduce that we need the new feature if  msg is longer then some value:
```
var SHORT_MSG_LENGTH = 120;
var shortMsg;

if (msg.length > SHORT_MSG_LENGTH) { ...
shortMsg = msg.substring(0, SHORT_MSG_LENGTH) + '...';
```

Reply:
thing is @himdel didn't want to hardcode the length thats why we added the options...

Reply:
It is still hard coded to `200` in [manageiq L94](https://github.com/ManageIQ/manageiq/pull/16550/files#diff-5915b4f8bc244828cd9b675fe145ee74R95) 

or am I missing something ?

Reply:
See the comments above...
> You're also randomly hardcoding a 200 character limit - this is wrong because this will affect any existing messages long enough. And it is also wrong because the reasonable size will depend on the user's resolution, and chosen language (this is why Twitter still limits Chinese to 140 chars).

Reply:
Currently the text in the line is too long ... so something is wrong here :-)
I would expect a flash to look like that:

Short view (only one or two lines in small screens)
```
Lorem ipsum dolor sit amet, per ne erat vitae sententiae, eam tollit sententiae ut, 
dicant eripuit suavitate an cum. Ea al...  Show More
```

Long view (show full message)
```
Lorem ipsum dolor sit amet, per ne erat vitae sententiae, eam tollit sententiae ut, 
dicant eripuit suavitate an cum. Ea alii elitr graecis has. Scripta iracundia cum eu, 
adipisci evertitur ea est. Mei in partem adversarium. Facete adipiscing necessitat
ibus qui at, an accusamus repudiare scripserit mea, odio dissentiunt ea sed. 

Show Less
```

@nimrodshn  @ohadlevy @himdel WDYT ?

Reply:
> Short view (only one or two lines in small screens)
@yaacov it still only show's one line, how short do you want the msg to be? half a line?

Reply:
@nimrodshn 
If `text-overflow: ellipsis;` does not work here (see gif in description) then you can go back to `trancate` by some constant, IMHO it's ok for this length to be more then one line on smaller screens, and it's also ok for this short version to be more short for some locales ( e.g. Chinese ), see [above example](https://github.com/ManageIQ/manageiq-ui-classic/pull/2862#discussion_r156909937).
@himdel WDYT ?

Reply:
> @yaacov it still only show's one line (regardless of the screen) , how short do you want the msg to be? half a line?

see [above example](https://github.com/ManageIQ/manageiq-ui-classic/pull/2862#discussion_r156909937).
If you look carefully you can see I show two lines in that example.

Reply:
> @yaacov it still only show's one line (regardless of the screen)

This is problematic on small devices, you sometimes want to show more then one line:

Big screen, one line is enough for short message: 
```
This is a problem that can be fixed easily by pressing this button
```

Small screen, you may want two or three lines for short message: 
```
This is a problem that
can be fixed easily by
pressing this button
```
If you show only one line:
```
This is a problem that...
```
This is not what I as user want to see :-)

Reply:
@yaacov Is this really a problem if the user can still expand the message to full length? Would your issue be solved if this was 2 lines instead of one?

As for https://github.com/ManageIQ/manageiq-ui-classic/pull/2862/files#r156909937 .. I'm not quite understanding what the issue is (or missing a screenshot where it goes wrong).